include/ginkgo/core/stop/residual_norm.hpp

```diff
@@ -226,6 +226,84 @@ class RelativeResidualNorm : public ResidualNorm<ValueType> {
 };
 
 
+/**
+ * The ImplicitResidualNormReduction class is a stopping criterion which
+ * stops the iteration process when the residual norm is below a certain
+ * threshold recurrent to the norm of the right-hand side, i.e. when
+ * norm(residual) / norm(right_hand_side) < threshold.
+ * For better performance, the checks are run thanks to kernels on
+ * the executor where the algorithm is executed.
+ *
+ * @note To use this stopping criterion there are some dependencies. The
+ * constructor depends on `b` in order to compute the norm of the
+ * right-hand side. If this is not correctly provided, an exception
+ * ::gko::NotSupported() is thrown.
```

Comment:
I don't understand what "recurrent to the norm of the right-hand side" means. 

I also think either this comment isn't correctly describing the class, or I am not understanding what the class is supposed to do (certainly possible!).  

My understanding is that what makes it an "implicit residual norm" class is that it's checking some value which is computed by the iterative solver -- the exact value being computed depends on the solver -- which we expect to tell us about the current residual, though it is not found by computing the norm of the residual itself.  For people familiar with other CG, GMRES, etc. implementations, this value is probably the one they're "used to" seeing as the value used for convergence checking.  Is this correct?  If so, I think the documentation should definitely mention this, so users realize this is likely the stopping criterion they want in order to replicate results they've gotten from other solvers.  

But why is it tied to the right hand side? The `ResidualNormReduction` class compares the residual to the initial residual, while the `RelativeResidualNorm` class compares to the rhs, which I feel makes this class's behavior confusing with respect to its name.  But also, can't we have both versions for the `ImplicitResidualNorm`?  One that compares to rhs and one that compares to initial residual?

Reply:
Yeah, I can do that. I have updated the factory to now take an additional parameter to select the comparison norm. I also updated the doc to be more clear.

Reply:
Cool, thanks!